src/odil/Reader.cpp

```diff
@@ -59,6 +59,8 @@ type value; \
 
 std::string read_string(std::istream & stream, unsigned int size)
 {
+    if(size == 0)
```

Comment:
Please avoid blocks without braces: the [goto fail](https://www.imperialviolet.org/2014/02/22/applebug.html) bug is still in many memories 😄